autoware_lanelet2_map_validator/src/validators/area/missing_regulatory_elements_for_bus_stop_areas.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2025 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lanelet2_map_validator/validators/area/missing_regulatory_elements_for_bus_stop_areas.hpp"
+
+#include "lanelet2_map_validator/utils.hpp"
+
+#include <range/v3/view/filter.hpp>
+
+#include <lanelet2_core/LaneletMap.h>
+
+#include <set>
+
+namespace lanelet::autoware::validation
+{
+namespace
+{
+lanelet::validation::RegisterMapValidator<MissingRegulatoryElementsForBusStopAreas> reg;
+}
+
+lanelet::validation::Issues MissingRegulatoryElementsForBusStopAreas::operator()(
+  const lanelet::LaneletMap & map)
+{
+  lanelet::validation::Issues issues;
+
+  lanelet::autoware::validation::appendIssues(
+    issues, check_missing_regulatory_elements_for_bus_stop_areas(map));
+
+  return issues;
+}
+
+lanelet::validation::Issues
+MissingRegulatoryElementsForBusStopAreas::check_missing_regulatory_elements_for_bus_stop_areas(
+  const lanelet::LaneletMap & map)
+{
+  lanelet::validation::Issues issues;
+
+  std::set<lanelet::Id> bus_stop_area_ids;
+
+  for (const auto & polygon : map.polygonLayer) {
+    const auto & attrs = polygon.attributes();
+    const auto & it = attrs.find(lanelet::AttributeName::Type);
+    if (it != attrs.end() && it->second == "bus_stop_area") {
+      bus_stop_area_ids.insert(polygon.id());
+    }
+  }
+
+  auto reg_elem_bus_stop = map.regulatoryElementLayer | ranges::views::filter([](auto && elem) {
+                             const auto & attrs = elem->attributes();
+                             const auto & it = attrs.find(lanelet::AttributeName::Subtype);
+                             return it != attrs.end() && it->second == "bus_stop_area";
```

Comment:
Please use lanelet::autoware::BusStopArea::RuleName instead of hard-coded "bus_stop_area", which is defined in autoware_lanelet2_extension